cmd-script.c

```diff
@@ -0,0 +1,181 @@
+/*
+ * Python script binding for function entry and exit
+ *
+ * Copyright (C) 2017, LG Electronics, Honggyu Kim <hong.gyu.kim@lge.com>
+ *
+ * Released under the GPL v2.
+ */
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <stdio_ext.h>
+
+#include "uftrace.h"
+#include "utils/utils.h"
+#include "utils/symbol.h"
+#include "utils/filter.h"
+#include "utils/fstack.h"
+#include "utils/script.h"
+
+#include "libtraceevent/event-parse.h"
+
+
+static int run_script_for_rstack(struct ftrace_file_handle *handle,
+				 struct ftrace_task_handle *task,
+				 struct opts *opts)
+{
+	struct uftrace_record *rstack = task->rstack;
+	struct uftrace_session_link *sessions = &handle->sessions;
+	struct sym *sym = NULL;
+	char *symname = NULL;
+
+	if (task == NULL)
+		return 0;
+
+	sym = task_find_sym(sessions, task, rstack);
+	symname = symbol_getname(sym, rstack->addr);
+
+	if (opts->kernel_skip_out) {
+		/* skip kernel functions outside user functions */
+		if (!task->user_stack_count && is_kernel_record(task, rstack))
+			goto out;
+	}
```

Comment:
I think it's better to move this block out of the function like `opts->kernel_only` checking.